pkg/systemd/reporting.jsx

```diff
@@ -0,0 +1,450 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2019 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import cockpit from "cockpit";
+import React from "react";
+import { show_modal_dialog } from "cockpit-components-dialog.jsx";
+
+const _ = cockpit.gettext;
+
+const TaskState = Object.freeze({
+    READY: 0,
+    RUNNING: 1,
+    COMPLETED: 2,
+    ERROR: 3,
+    CANCELED: 4,
+});
+
+const PromptType = Object.freeze({
+    ASK: 0,
+    ASK_YES_NO: 1,
+    ASK_YES_NO_YESFOREVER: 2,
+    ASK_YES_NO_SAVE: 3,
+    ASK_PASSWORD: 4,
+});
+
+const ProblemState = Object.freeze({
+    REPORTABLE: 0,
+    REPORTING: 1,
+    REPORTED: 2,
+    UNREPORTABLE: 3,
+});
+
+// Avoiding using the proxy to circumvent the GLib cache.
```

Comment:
This needs a more detailled explanation. A dbus proxy does react to property updates. However, if you only need to get one or two properties from each object, then a proxy is indeed rather heavyweight, and a raw cockpit.dbus() is more efficient. That would be a legit reason, but please adjust the comment accordingly if that's it.

Reply:
It’s been months since I last looked into this, but I believe the change notifications weren’t coming and the proxy still held old values. I tracked it down to somewhere in GLib, but couldn’t tell you anything concrete at this point.